functions.py

```diff
@@ -0,0 +1,60 @@
+import requests
+from bs4 import BeautifulSoup as bs
+import json
+
+URL = "https://www.producthunt.com/"
+
+response = requests.get(URL)
+
+html = bs(response.content, 'lxml')
+
+
+tags = html.find('script', attrs = {'id':"__NEXT_DATA__"}).string.strip()
+
+
+data = json.loads(tags)
+
+
+data = data["props"]["apolloState"]
+
+ref = []  #ref will have product ids (not as it is)
+
+for i in data:
+        if "HomefeedPageFEATURED" in i:
+            for j in range(len(data[i]["items"])):
+                ref.append(data[i]["items"][j]["__ref"])
+
+upvote_list = []

```

Comment:
good name.Here I know this variable is of type list and will hold stuff related to upvotes